app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,90 @@
+class OrdersController < ApplicationController
+  before_action :require_login, only: [:index]
+  before_action :correct_user, only: [:index]
+  before_action :order_items, only: [:new, :create]
+  before_action :find_user, except: [:new, :create, :confirm, :cancel]
+
+  def index
+    orders = @user.orders
+    @orders = orders.uniq.sort
+    @paid_orders = Order.filter_orders(@orders, "paid")
+    @completed_orders = Order.filter_orders(@orders, "shipped")
+    @cancelled_orders = Order.filter_orders(@orders, "cancelled")
+    @paid_orders_revenue = Order.user_orders_revenue(@paid_orders, @user)
+    @completed_orders_revenue = Order.user_orders_revenue(@completed_orders, @user)
+    @total_revenue = @paid_orders_revenue + @completed_orders_revenue
+    if !(params["Status"].nil? || params["Status"] == "")
+      @orders = Order.filter_orders(@orders, params["Status"])
+    end
+  end
+
+  def show
+    @order = Order.find(params[:id])
+  end
+
+  def new
+    not_enough_stock = false
+    @order_items.each do |oi|
+      not_enough_stock = true if Product.find(oi.product_id).stock < oi.quantity
+    end
+    if not_enough_stock
+      flash[:error] = "There is not enough stock to complete your purchase. Please update your cart."
```

Comment:
Ideally we would indicate to our user which items do not have sufficient stock to create the order. How might we refactor this code to support that?